src/store/mutations/mutations-model.ts

```diff
@@ -259,8 +260,8 @@ export const mutateProvisionsRemoved = (state: StateIF, provisionsRemoved: boole
   state.stateModel.newAlteration.provisionsRemoved = provisionsRemoved
 }
 
-export const mutateOriginalResolutionDates = (state: StateIF, resolutionDates: string[]) => {
-  state.stateModel.originalAlteration.alteration.shareStructure.resolutionDates = resolutionDates
+export const mutateOriginalResolutionDates = (state: StateIF, resolutionDates: ResolutionsIF[]) => {
+  state.stateModel.shareStructureStep.previousResolutionDates = resolutionDates
```

Comment:
Is there any chance that `resolutionDates` is null?
If so, should provide fallback to [], or use conditional chaining everywhere it's used to prevent de-referencing a null array. (eg, list[0]?.field)

Reply:
In this case, this is defined in the store as an empty [].

Reply:
Yes, but do we ever assign anything else to it, like null?

Reply:
Should only ever be an empty array. I don't believe we would assign it `null`. 
Obviously i can't guarantee  nothing could/would ever go wrong but i do know how the component handles these dates. 
It either adds or subtracts strings(dates) from this array but never wipes the [] afaik.